WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginSiteAddressFragment.java

```diff
@@ -198,7 +208,42 @@ protected void discover() {
         }
 
         mRequestedSiteAddress = getCleanedSiteAddress();
-        mDispatcher.dispatch(AuthenticationActionBuilder.newDiscoverEndpointAction(mRequestedSiteAddress));
+
+        Uri uri = Uri.parse(UrlUtils.addUrlSchemeIfNeeded(mRequestedSiteAddress, false));
+        String endpoint = WPCOMREST.sites.getUrlV1() + uri.getHost();
+        WPComGsonRequest req = WPComGsonRequest.buildGetRequest(endpoint, null, SiteWPComRestResponse.class,
+                new Response.Listener<SiteWPComRestResponse>() {
+                    @Override
+                    public void onResponse(SiteWPComRestResponse response) {
+                        if (response.jetpack) {
```

Comment:
That's actually the expected behaviour. Since `nbradbury.com` is not a (mapped) WordPress.com site, the request will fail and be handled in `onErrorResponse` as you've already observed.

If you notice the comment on [the following line](https://github.com/wordpress-mobile/WordPress-Android/pull/6089/files#diff-bbc95f0597a9d70e9bb0293ddc4c0122R219), the Jetpack condition is there to force the WordPress.com site to be treated as selfhosted instead of WPCOM. That way, user's selfhosted credentials will be accepted and the selfhosted network calls (XMLRPC) will be used. Let me know if that makes this more clear.